autoPyTorch/evaluation/tae.py

```diff
@@ -0,0 +1,433 @@
+                    status = StatusType.TIMEOUT
+                    additional_run_info = {'error': 'Timeout'}
+                elif obj.exit_status is pynisher.MemorylimitException:
+                    status = StatusType.MEMOUT
+                    additional_run_info = {
+                        'error': 'Memout (used more than %d MB).' % self.memory_limit
+                    }
+                else:
+                    raise ValueError(obj.exit_status)
+                cost = self.worst_possible_result
+
+        elif obj.exit_status is TAEAbortException:
+            info = None
+            status = StatusType.ABORT
+            cost = self.worst_possible_result
+            additional_run_info = {'error': 'Your configuration of '
+                                            'autoPyTorch does not work!',
+                                   'exit_status': _encode_exit_status(obj.exit_status),
+                                   'subprocess_stdout': obj.stdout,
+                                   'subprocess_stderr': obj.stderr,
+                                   }
+
+        else:
+            try:
+                info = read_queue(queue)
+                result = info[-1]['loss']
+                status = info[-1]['status']
+                additional_run_info = info[-1]['additional_run_info']
+
+                if obj.exit_status == 0:
+                    cost = result
+                else:
+                    status = StatusType.CRASHED
+                    cost = self.worst_possible_result
+                    additional_run_info['info'] = 'Run treated as crashed ' \
+                                                  'because the pynisher exit ' \
+                                                  'status %s is unknown.' % \
+                                                  str(obj.exit_status)
+                    additional_run_info['exit_status'] = _encode_exit_status(obj.exit_status)
+                    additional_run_info['subprocess_stdout'] = obj.stdout
+                    additional_run_info['subprocess_stderr'] = obj.stderr
+            except Empty:
+                info = None
+                additional_run_info = {
+                    'error': 'Result queue is empty',
+                    'exit_status': _encode_exit_status(obj.exit_status),
+                    'subprocess_stdout': obj.stdout,
+                    'subprocess_stderr': obj.stderr,
+                    'exitcode': obj.exitcode
+                }
+                status = StatusType.CRASHED
+                cost = self.worst_possible_result
+
+        if (
+                (self.budget_type is None or budget == 0)
+                and status == StatusType.DONOTADVANCE
+        ):
+            status = StatusType.SUCCESS
+
+        if not isinstance(additional_run_info, dict):
+            additional_run_info = {'message': additional_run_info}
+
+        if (
+                info is not None
```

Comment:
We would have to do this in all cases in AutoPyTorch right? As all the fits would be iterative